akka-docs/src/main/paradox/stream/stream-flows-and-basics.md

```diff
@@ -421,13 +421,13 @@ Scala
 Java
 :   @@snip [FlowDocTest.java](/akka-docs/src/test/java/jdocs/stream/FlowDocTest.java) { #materializer-from-system-in-actor }
 
-In the above example we pass in a materializer to the Actor, which results in binding its lifecycle to the entire `ActorSystem` rather than the single enclosing actor. This can be useful if you want to share a materializer or group streams into specific materializers,
+In the above example we pass in a materializer to the Actor, which results in binding its lifecycle to the entire @apidoc[akka.actor.ActorSystem] rather than the single enclosing actor. This can be useful if you want to share a materializer or group streams into specific materializers,
 for example because of the materializer's settings etc.
 
 @@@ warning
 
-Do not create new actor materializers inside actors by passing the `context.system` to it. 
-This will cause a new `Materializer` to be created and potentially leaked (unless you shut it down explicitly) for each such actor.
+Do not create new actor materializers inside actors by passing the @scala[@scaladoc[context.system](akka.actor.ActorContext#system:akka.actor.ActorSystem)]@java[@javadoc[context.system()](akka.actor.ActorContext#system())] to it. 
+This will cause a new @apidoc[akka.stream.Materializer] to be created and potentially leaked (unless you shut it down explicitly) for each such actor.
 It is instead recommended to either pass-in the Materializer or create one using the actor's `context`.
```

Comment:
Should I add the link to `context`? Which link?

Reply:
I think we can leave it just keyword (it could be both typed actor context and classic actor context)